jpa/odata-jpa-processor/src/main/java/org/apache/olingo/jpa/processor/core/query/EntityQueryBuilder.java

```diff
@@ -153,8 +159,10 @@ protected final QueryEntityResult executeInternal(final boolean processExpandOpt
     queryResult.putElementCollectionResults(readElementCollections(elementCollectionMap));
 
     if (processExpandOption && !intermediateResult.isEmpty()) {
+      String idAttributeName = getQueryResultType().getKeyAttributes(false).get(0).getInternalName();
```

Comment:
This line is the reason for some of the test failures: your using the attribute name to get the result value from the tuple, but in the tuple the result is accessible only be the alias... so you have to do:
_String idAttributeName = getLastAffectingNavigationKeyBuilder().getNavigationKeyPaths().get(0).getAlias();_
But there are more issues... one should be the fact, that you using only one column as key (see ...get(0)), but there are entities with multiple key columns in the world, so you have to bring ALL key columns into effect for the expand query...

Reply:
Hi
What would be the proper way to get all the key fields?
Anyway, I think not handling multi-column keys is still better than it was before :)

Reply:
`getLastAffectingNavigationKeyBuilder().getNavigationKeyPaths().get(0).getAlias()` gives me something else and not a key column name. My JPA entities does not work with this change

Reply:
It seems we cannot use tuples in JPA criteria builder of an inClause. Anyway odata cannot express composite primary keys either